helion/_compiler/tile_strategy.py

```diff
@@ -433,7 +441,9 @@ def _setup_mask(
             f"{mask_var} = ({index_var} < ({state.device_function.sympy_expr(numel)}))"
         )
 
-    def select_pid_strategy(self) -> ProgramIDs:
+    def select_pid_strategy(self, state: CodegenState) -> ProgramIDs:
+        if (shared_pid := state.device_function.shared_pid) is not None:
+            return shared_pid
         if self.l2_grouping > 1:
             return L2GroupingProgramIDs(group_size=self.l2_grouping)
         if 1 < len(self.block_indices) <= 3 and self.fn.config.use_yz_grid:
```

Comment:
Are we setting updating the config spec so `use_yz_grid` is removed from the config?